### PR TITLE
Scripts/Shadowmoon Valley: Dragonmaw Peon AI Update

### PR DESCRIPTION
**Pull request by ghost**
Fixes issue #16210.

### CODE REVIEW

**Comment by Saben65** [comment]
Seems to be some additional issues with the scripting. The NPCs will stay aggro'd to you indefinitely without evading due to distance. If you do drop aggro (I could only do so by enabling GM mode) they will stay wherever they dropped aggro and do their mining emote instead of running back to their spawn point.

Not sure how to fix this.

**Comment by MitchesD** [comment]
you should check for victim

**Comment by Saben65** [comment]
I assume you are talking about this?

```C++
            if (!UpdateVictim())
                return;
```

That fixed the issue where they wouldn't reset.

File: src/server/scripts/Outland/zone_shadowmoon_valley.cpp [hunk 1]
```diff
@@ -514,6 +514,8 @@ class npc_dragonmaw_peon : public CreatureScript
             Initialize();
         }
 
+        void EnterCombat(Unit* /*who*/) override { }
```
Comment (jackpoz) [inline src/server/scripts/Outland/zone_shadowmoon_valley.cpp]: is this changed needed for a particular reason ?

Comment (Saben65) [inline src/server/scripts/Outland/zone_shadowmoon_valley.cpp]: Nothing in particular. I just added it based on it being there in similar scripts. I will remove it.

**Comment by jackpoz** [comment]
I aggroed 2, moved a bit away, died with ".die" and 1 of the peons moved to his position while the other did the mining animation for a few seconds and then moved to his position. Not sure if there's a way to make them do the mining position only after going back to their position

**Pull request closed without merging**